code/datums/components/itempicky.dm

```diff
@@ -5,13 +5,23 @@
 	var/whitelist
 	/// Message shown if you try to pick up an item not in the whitelist
 	var/message = "You don't like %TARGET, why would you hold it?"
+	/// An optional condition we check for overriding our whitelist;
+	/// can be a var or a callback
+	var/tertiary_condition = null
 
-/datum/component/itempicky/Initialize(whitelist, message)
+/datum/component/itempicky/Initialize(whitelist, message, tertiary_condition)
 	if(!ismob(parent))
 		return COMPONENT_INCOMPATIBLE
 	src.whitelist = whitelist
 	if(message)
 		src.message = message
+	if(tertiary_condition)
+		if(istype(tertiary_condition, /datum/callback))
+			// if we're tracking some value that we won't be changing and
+			// isn't resolved logically by a callback, we'll resolve our reference to it instead
+			src.tertiary_condition = tertiary_condition
+		else
+			src.tertiary_condition = &tertiary_condition
```

Comment:
do not use pointers in this manner (maybe at all) I will kill you with hammers. they lead to potentially hung refs and have no meaningful upside in this usecase. it is also real poor form to mix typing on vars like this

Reply:
also, how would this even work? you're passing in a var as an arg, then converting THIS VARIABLE into a pointer, so the moment this proc ends it'll point at nothing useful?